### PR TITLE
feat(nodejs): custom logging

### PR DESCRIPTION
**Pull request by glasstiger**
Extended the options of the client with a new property: `log`.
Now the host application can pass a logging function to the client.
```
log: (level: 'error'|'warn'|'info'|'debug', message: string) => void
```
If this property not provided, default logging is used which writes to the console with logging level `info`.

### CODE REVIEW

File: src/sender.js [hunk 1]
```diff
@@ -155,15 +162,15 @@ class Sender {
     /**
      * Closes the connection to the database. <br>
      * Data sitting in the Sender's buffer will be lost unless flush() is called before close().
+     *
+     * @return {Promise<boolean>} Resolves to true if connection is closed.
```
Comment (puzpuzpuz) [inline src/sender.js:166]: nit: why do we need to return anything here, especially considering that the returned value is always `true`?

File: types/src/sender.d.ts [hunk 2]
```diff
@@ -70,7 +73,7 @@ export class Sender {
      * Closes the connection to the database. <br>
      * Data sitting in the Sender's buffer will be lost unless flush() is called before close().
      */
-    close(): Promise<any>;
+    close(): Promise<void>;
```
Comment (puzpuzpuz) [inline types/src/sender.d.ts:76]: Return type is not consistent with the actual one which is boolean.

**Review by puzpuzpuz — APPROVED** [review]
LGTM. Left a couple minor comments.

File: src/sender.js (hunk 1 above)
Comment (glasstiger) [reply src/sender.js:166]: ok, removing the return value.

File: types/src/sender.d.ts (hunk 2 above)
Comment (glasstiger) [reply types/src/sender.d.ts:76]: yes, i generated this before i added the `return true;` line.
will regenerate everything, just in case.

Comment (glasstiger) [reply types/src/sender.d.ts:78]: removed `return true;`, it is consistent now

File: src/sender.js (hunk 1 above)
Comment (glasstiger) [reply src/sender.js]: removed `return true;`, i think it looks better now.

**Pull request merged**